### PR TITLE
[Plugin] Make the wasi-logging as a built-in plugin.

### PR DESCRIPTION
**Pull request by q82419**

### CODE REVIEW

**Comment by juntao** [comment]
Hello, I am a [code review agent](https://github.com/flows-network/github-pr-review/) on [flows.network](https://flows.network/).

It could take a few minutes for me to analyze this PR. Relax, grab a cup of coffee and check back later. Thanks!

File: .github/workflows/matrix-extensions.json [hunk 1]
```diff
@@ -170,4 +159,4 @@
             "options": "-DWASMEDGE_PLUGIN_ZLIB=ON"
         }
     }
-}
+}
```
Comment (hydai) [inline .github/workflows/matrix-extensions.json:162]: EOF

**Review by hydai — CHANGES_REQUESTED** [review]

File: lib/plugin/plugin.cpp [hunk 2]
```diff
@@ -48,7 +52,7 @@ namespace WasmEdge {
 namespace PO {
 template <> struct Parser<WasmEdge_String> {
   static cxx20::expected<WasmEdge_String, Error>
-  parse(std::string Value) noexcept {
+  parse(std::string_view Value) noexcept {
```
Comment (hydai) [inline lib/plugin/plugin.cpp:55]: We should keep `std::string`. Ref: `include/po/parser.h`

File: test/plugins/wasi_logging/wasi_logging.cpp [hunk 3]
```diff
@@ -18,9 +16,7 @@ namespace {
 
 WasmEdge::Runtime::Instance::ModuleInstance *createModule() {
   using namespace std::literals::string_view_literals;
-  WasmEdge::Plugin::Plugin::load(std::filesystem::u8path(
-      "../../../plugins/wasi_logging/" WASMEDGE_LIB_PREFIX
-      "wasmedgePluginWasiLogging" WASMEDGE_LIB_EXTENSION));
+  WasmEdge::Plugin::Plugin::loadFromDefaultPaths();
```
Comment (hydai) [inline test/plugins/wasi_logging/wasi_logging.cpp:19]: We should add a comment to identify that `built-in plugin` is loaded in this function.

File: test/plugins/wasi_logging/wasi_logging.cpp [hunk 4]
```diff
@@ -58,68 +58,122 @@ TEST(WasiLoggingTests, func_log) {
   auto &MemInst = *MemInstPtr;
   WasmEdge::Runtime::CallingFrame CallFrame(nullptr, &Mod);
 
-  // Clear the memory[0, 32].
-  fillMemContent(MemInst, 0, 32);
-  // Set strings in memory
+  // Clear the memory[0, 64].
```
Comment (hydai) [inline test/plugins/wasi_logging/wasi_logging.cpp:61]: Number mismatched.

File: test/plugins/wasi_logging/wasi_logging.cpp [hunk 5]
```diff
@@ -58,68 +58,122 @@ TEST(WasiLoggingTests, func_log) {
...
-  EXPECT_NE(FuncInst, nullptr);
-  EXPECT_TRUE(FuncInst->isHostFunction());
-  auto &HostFuncInst =
-      dynamic_cast<WasmEdge::Host::WASILogging::Log &>(FuncInst->getHostFunc());
+  fillMemContent(MemInst, 16, std::string("out.log"));
+  fillMemContent(MemInst, 24, std::string("out2.log"));
+  fillMemContent(MemInst, 128, std::string("This is log message"));
+  fillMemContent(MemInst, 160, std::string("Message 1 to file"));
+  fillMemContent(MemInst, 192, std::string("Message 2 to file"));
+  fillMemContent(MemInst, 224, std::string("Message 3 to file"));
```
Comment (hydai) [inline test/plugins/wasi_logging/wasi_logging.cpp:71]: Let's try `std::string_view`.

**Review by hydai — CHANGES_REQUESTED** [review]

**Review by hydai — APPROVED** [review]
LGTM.

**Comment by hydai** [comment]
Let's wait for CI pass.

**Review by hydai — APPROVED** [review]

**Pull request merged**